app/models/supported_ssg.rb

```diff
@@ -89,9 +88,21 @@ def os_version(rhel)
       rhel.scan(/(\d+)\.(\d+)$/)[0]
     end
 
+    def clear
+      @raw_supported = nil
+      @checksum = ConfigDownloader.ssg_ds_checksum
+    end
+
+    def ssg_ds_changed?
+      ConfigDownloader.update_ssg_ds
+
+      @checksum != ConfigDownloader.ssg_ds_checksum
+    end
+
     def raw_supported
-      # cached for the whole runtime
-      @raw_supported ||= YAML.load_file(self::SUPPORTED_FILE)
+      clear if ssg_ds_changed?
```

Comment:
It looks like that every time any general `SupportedSsg` method would be called it would re-download the file, is that right? Can it be cached at least for the duration of the request or time cached?

Reply:
It definitely downloads on every method call, as advertised:

> the config checksum is compared every time SupportedSsg is used, but the yaml is not parsed every time

I don't want to prematurely optimize this, especially since we can go back and add it easily later. Note that remediations also downloads content on every request from compliance-ssg - it really shouldn't be a problem. I am also wary of caching and wouldn't add it unless it is necessary.

That said, I think the best way to improve this current design is to serve the checksums from compliance-ssg so very little work (or network) is necessary to determine if the config has changed. I don't see why we can't merge this PR and improve it in the meantime.